app/controllers/admin/site_settings_controller.rb

```diff
@@ -9,40 +9,71 @@ def update
     authorise Setting.all.first
 
     all_updated = update_settings( settings_params )
-    if all_updated
+    if all_updated.nil?
+      flash[ :alert_info ] = t( '.unchanged' )
+    elsif all_updated
       flash[ :notice ] = t( '.success' )
     else
       flash[ :alert ] = t( '.failure' )
     end
-    redirect_to site_settings_path
+    redirect_to admin_site_settings_path
   end
 
   private
 
   def update_settings( params )
-    flag = true
+    flags = initialise_flags
     params&.each_key do |key|
       next unless /value_(?<id>\d+)/ =~ key
 
-      flag = false unless update_setting( id, params )
+      flag  = update_setting( id, params )
+      flags = update_flags( flag, flags )
     end
-    flag
+    return if flags[ :no_change ]
+
+    flags[ :no_errors ]
   end
 
   def update_setting( id, params )
     setting = Setting.find( id )
     authorise setting
-    level = params[ "level_#{id}" ]
 
-    unless level.blank? || setting.level == level
-      return false unless setting.update( level: level )
-    end
+    level_flag = update_level( setting, params )
+    value_flag = update_value( setting, params )
+
+    return if level_flag.nil? && value_flag.nil?
+    return false if level_flag == false || value_flag == false
+
+    true
+  end
+
+  def update_level( setting, params )
+    level_param = params[ "level_#{setting.id}" ]
+
+    return if level_param.blank? || setting.level == level_param
 
-    val = setting.values.find_by( user_id: nil )
-    value = params[ "value_#{id}" ] || ''
-    return true if val.value == value
+    _ret = setting.update( level: level_param )
+  end
+
+  def update_value( setting, params )
+    value_param = params[ "value_#{setting.id}" ] || ''
+
+    value = setting.values.find_by( user_id: nil )
+
+    _ret = value.update( value: value_param ) unless value.value == value_param
+  end
+
+  def initialise_flags
+    flags = {}
+    flags[ :no_change ] = true
+    flags[ :no_errors ] = true
+    flags
+  end
 
-    val.update!( value: value )
+  def update_flags( flag, flags )
+    flags[ :no_change ] = false unless flag.nil?
+    flags[ :no_errors ] = false if flag == false
+    flags
```

Comment:
I'm finding some of the methods in this controller a little tricky to follow. 

However I admit I've not got a better suggestion.

It *may* be that using the `ActiveModel::Dirty` methods like `#changed?` can help simplify the code here, but no guarantees: https://api.rubyonrails.org/classes/ActiveModel/Dirty.html